### DISCUSSION TITLE
How to customize the chart query

### DISCUSSION THREAD

**Discussion by nagarajmmu** [apache/superset#28748] [Q&A / Help]
Hi All

We are using superset as our visualization tool, we are analyzing the big data, so charts are loading very slow.
but we are able to increase the performance by selecting only the selected column. But superset is using a query as below.
```
SELECT "ID" AS "ID",
       "YEAR" AS "YEAR",
       "AMOUNT" AS "AMOUNT",
       "CREDIT" AS "CREDIT",
       count(DISTINCT "SOURCE") AS "COUNT_DISTINCT(SOURCE)"
FROM
  (select *  from MY_Table) AS virtual_table
GROUP BY "ID",
         "YEAR",
         "AMOUNT",
         "CREDIT"
ORDER BY "COUNT_DISTINCT(SOURCE)" DESC
LIMIT 100;
```
In above query, superset is selecting all the columns (**select *  from MY_Table**) and grouping.
Is there any way we can customize the chart queries as we needed.
Please let us know, it would be great help.

Thanks
Nagaraj M M

**Comment by sarcoex** [comment]
still struggling with this
